pkg/reconciler/vertex/scaling/scaling.go

```diff
@@ -223,7 +223,13 @@ func (s *Scaler) scaleOneVertex(ctx context.Context, key string, worker int) err
 		return err
 	}
 
-	if vertex.Status.Replicas == 0 { // Was scaled to 0
+	if vertex.Status.Replicas == 0 {
```

Comment:
Move to vertex.GetReplicas()

Reply:
Or similar to mvtx, to have a Calculate Replicas()

Reply:
In this way, don't need to make change to scaling.

Reply:
Sure will take a look. Thanks!

Reply:
I had a discussion with Sidhant. We need to figure out why at the first place we have a vertex with 0 replica instead of having this hardcoded check. Looking at the code it's hard to figure out. I am creating an issue to track this.